### PR TITLE
Makes `client.images` respect changes

### LINKED ISSUE #1422: Images added to `client.images` do not update their appearance.
Once an `/image` is added to `client.images` its appearance is static and changes to its vars do not reflect on the client.

### PR DESCRIPTION
**Pull request by amylizzle**
Creates a temporary entity for the image, attaches the appearance to that.

Fixes #1422  

~~Minor concern is that entities are too expensive though.~~ No they're not, tested on Paradise as AI and it was smooth as butter.

### CODE REVIEW

**Comment by amylizzle** [comment]
Note: tested on paradise as AI, performance is great.

File: OpenDreamClient/Rendering/ClientImagesSystem.cs [hunk 1]
```diff
@@ -69,17 +52,21 @@ internal sealed class ClientImagesSystem : SharedClientImagesSystem {
...
-                if(!_turfClientImages.TryGetValue(e.TurfCoords, out var iconList))
-                    return;
-                iconList.Remove(e.ImageAppearance);
+            if(!_turfClientImages.TryGetValue(e.TurfCoords, out var iconList))
+                return;
+            iconList.Remove(e.ImageEntity);
+            if(iconList.Count == 0)
+                _turfClientImages.Remove(e.TurfCoords);
+            else
                 _turfClientImages[e.TurfCoords] = iconList;
```
Comment (wixoaGit) [inline OpenDreamClient/Rendering/ClientImagesSystem.cs:61]: This should be unnecessary. `_turfClientImages[e.TurfCoords]` is already a reference to `iconList`.

File: OpenDreamClient/Rendering/ClientImagesSystem.cs [hunk 2]
```diff
@@ -69,17 +52,21 @@ internal sealed class ClientImagesSystem : SharedClientImagesSystem {
...
             if(!_amClientImages.TryGetValue(ent, out var iconList))
                 return;
-            iconList.Remove(e.ImageAppearance);
-            _amClientImages[ent] = iconList;
-            _idToIcon.Remove(e.ImageAppearance);
+            iconList.Remove(e.ImageEntity);
+            if(iconList.Count == 0)
+                _amClientImages.Remove(ent);
+            else
+                _amClientImages[ent] = iconList;
```
Comment (wixoaGit) [inline OpenDreamClient/Rendering/ClientImagesSystem.cs:69]: This should also be unnecessary for the same reason.

File: OpenDreamRuntime/Objects/Types/DreamObjectImage.cs [hunk 3]
```diff
@@ -1,14 +1,18 @@
 ﻿using OpenDreamRuntime.Procs;
+using OpenDreamRuntime.Rendering;
 using OpenDreamShared.Dream;
+using Robust.Shared.Map;
 
 namespace OpenDreamRuntime.Objects.Types;
 
 public sealed class DreamObjectImage : DreamObject {
+
```
Comment (wixoaGit) [inline OpenDreamRuntime/Objects/Types/DreamObjectImage.cs:9]:
```suggestion
```

**Comment by amylizzle** [comment]
Can't replicate that weird effect at all. Can you still?

**Comment by wixoaGit** [comment]
> Can't replicate that weird effect at all. Can you still?

I cannot. Appears to be a one-off unrelated graphical glitch.

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**